src/movies.js

```diff
@@ -1,16 +1,150 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(array) {
+  let directors = array.map((movie) => {
+    return movie.director;
+  });
+  return directors;
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array) {
+  let spielbergMovie = array.filter((movie) => {
+    return (
+      movie.director === "Steven Spielberg" &&
+      movie.genre.indexOf("Drama") != -1
+    );
+  });
+  return spielbergMovie.length;
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(array) {
+  if (array.length !== 0) {
+    let total = array.reduce((acc, movie) => {
+      if (movie.rate != "" && movie.rate != undefined) {
+        return acc + movie.rate;
+      } else {
+        return acc;
+      }
+    }, 0);
+    total = total / array.length;
+    return Number(total.toFixed(2));
+  } else {
+    return 0;
+  }
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
-// Iteration 5: Ordering by year - Order by year, ascending (in growing order)
+function dramaMoviesRate(array) {
+  let dramaMovies = array.filter((movie) => {
+    return (
+      movie.rate != "" &&
+      movie.rate != undefined &&
+      movie.genre.indexOf("Drama") != -1
+    );
+  });
+
+  if (dramaMovies.length !== 0) {
+    let total = dramaMovies.reduce((acc, movie) => {
+      if (dramaMovies.length > 0) {
```

Comment:
this condition is already being checked in line 50 so there is no need to run it again.